### PR TITLE
implement FromStr for DataRate enum

### PR DESCRIPTION
**Pull request by michaeldjeffrey**
Questions:
- Is DecodeError implemented properly?
- An easier way to compare strings?
- Is the impl block scoped correctly?

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -43,4 +44,39 @@ pub mod services {
...
+                "SF8BW500" => Ok(DataRate::Sf8bw500),
+                "SF7BW500" => Ok(DataRate::Sf7bw500),
+                "LRFHSS1BW137" => Ok(DataRate::Lrfhss1bw137),
+                "LRFHSS2BW137" => Ok(DataRate::Lrfhss2bw137),
+                "LRFHSS1BW336" => Ok(DataRate::Lrfhss1bw336),
+                "LRFHSS2BW336" => Ok(DataRate::Lrfhss2bw336),
+                "LRFHSS1BW1523" => Ok(DataRate::Lrfhss1bw1523),
+                "LRFHSS2BW1523" => Ok(DataRate::Lrfhss2bw1523),
+                "FSK50" => Ok(DataRate::Fsk50),
+                unknown => Err(prost::DecodeError::new(format!("unknown datarate: {}", unknown)))
```
Comment (madninja) [inline src/lib.rs:78]: You can put the unknown variable in the placeholder of the format call

**Comment by madninja** [comment]
Technically the impl doesn’t have to be in the services section at all but it doesn’t hurt either

**Comment by madninja** [comment]
There are tactics for a perfect parser given the known set but this is fine for now I think

File: src/lib.rs (hunk 1 above)
Comment (michaeldjeffrey) [reply src/lib.rs]: Tried that before I added `format!` and forgot to try it again when I did. Thank you

**Pull request merged**